src/cargo/ops/cargo_rustc/context.rs

```diff
@@ -143,36 +143,40 @@ impl<'a, 'cfg> Context<'a, 'cfg> {
 
     /// Prepare this context, ensuring that all filesystem directories are in
     /// place.
-    pub fn prepare(&mut self, pkg: &'a Package,
-                   targets: &[(&'a Target, &'a Profile)])
-                   -> CargoResult<()> {
+    pub fn prepare(&mut self, root: &Package,
+                   pkgs: &'a PackagesToBuild<'a>) -> CargoResult<()> {
         let _p = profile::start("preparing layout");
 
         try!(self.host.prepare().chain_error(|| {
             internal(format!("couldn't prepare build directories for `{}`",
-                             pkg.name()))
+                             pkgs.iter().map(|&(ref pkg, _)| format!("`{}`", pkg.name()))
+                                        .collect::<Vec<_>>().join(",")))
```

Comment:
It's ok to just omit the names of the packages here, "couldn't prepare build directories" should be good enough (same below as well)